### ISSUE TITLE
AmplifyPushNotificationsPinpoint.onNotificationOpened not worked for IOS

### ISSUE THREAD

**Issue by windcludy** [aws-amplify/amplify-flutter#3273]
### Description

I'm using AmplifyPushNotificationsPinpoint.onNotificationOpened to listen for app open event from push notification。
 It works well on Android. but It does not worked for iOS 。

when open app by push notification a warning is appeared。

Warning: UNUserNotificationCenter delegate received call to -userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler: but the completion handler was never called.

How can I fixed this

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [ ] DataStore
- [X] Notifications (Push)
- [ ] Storage

### Platforms

- [X] iOS
- [ ] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.10.2

### Amplify Flutter Version

1.1.0

### Deployment Method

Amplify CLI

**Comment by HuiSF** [comment]
Hi @windcludy sorry to hear that you are experiencing this issue.

> Warning: UNUserNotificationCenter delegate received call to -userNotificationCenter:didReceiveNotificationResponse:withCompletionHandler: but the completion handler was never called.

Currently Amplify Flutter Push Notification plugin doesn't use `UNUserNotificationCenter` APIs to implement PN related functions. Did you make any custom implementation that uses `UNUserNotificationCenter` in your AppDelegate by any chance?

**Comment by windcludy** [comment]
Hi @HuiSF  thanks for your answer。
I have not  make any custom，everything is default

my source：

`import UIKit
import Flutter
import flutter_local_notifications

@UIApplicationMain
@objc class AppDelegate: FlutterAppDelegate {
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
    // GeneratedPluginRegistrant.register(with: self)
    // return super.application(application, didFinishLaunchingWithOptions: launchOptions)

    // This is required to make any communication available in the action isolate.
    FlutterLocalNotificationsPlugin.setPluginRegistrantCallback { (registry) in
        GeneratedPluginRegistrant.register(with: registry)
    }

    if #available(iOS 10.0, *) {
      UNUserNotificationCenter.current().delegate = self as UNUserNotificationCenterDelegate
    }

    GeneratedPluginRegistrant.register(with: self)
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }

}
`

**Comment by HuiSF** [comment]
Thanks for the source code @windcludy I see this line 
```swift
UNUserNotificationCenter.current().delegate = self as UNUserNotificationCenterDelegate
```
Is assigning your AppDelegate as the delegate of `UNUserNotificationCenter`, but you haven't implemented any of the methods. Please try to remove the entire block and rebuild the App and test the PN see if it works for you.


```swift
    if #available(iOS 10.0, *) {
      UNUserNotificationCenter.current().delegate = self as UNUserNotificationCenterDelegate
    }
```

**Comment by windcludy** [comment]
Hi @HuiSF 

when remove the  entire block , onNotificationOpened is called by open message.
thank you for your answer.

```
if #available(iOS 10.0, *) {
      UNUserNotificationCenter.current().delegate = self as UNUserNotificationCenterDelegate
    }
```

**Comment by buddy-bas** [comment]
Hi @HuiSF 

I want to display notification when app state is foreground so I add 
```
if #available(iOS 10.0, *) {
      UNUserNotificationCenter.current().delegate = self as UNUserNotificationCenterDelegate
    }
```
in appDelegate but it conflict with onNotificationOpened what should I do ?

**Comment by HuiSF** [comment]
Hi @buddy-bas current the Push Notification plugin does not support PN presentation related feature. There is a [feature request](https://github.com/aws-amplify/amplify-flutter/issues/3127) for this. Please plus on on this FR, and if you can, please share a description of the use cases you desire. Thanks!
